server/src/main/java/org/opensearch/index/store/remote/filecache/NonBlockCachedIndexInput.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.remote.filecache;
+
+import org.apache.lucene.store.IndexInput;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of the CachedIndexInput for NON_BLOCK files which takes in an IndexInput as parameter
+ */
+public class NonBlockCachedIndexInput implements CachedIndexInput {
```

Comment:
probably using `FullFile` is more explicit vs `NonBlock` which could mean anything.